### PR TITLE
[BB-4015] Monitor active users WIP

### PR DESCRIPTION
**Pull request by janimo**
## Description

This adds a task similar to the existing monthly report. It sends an email with the number of logged in users for the past month for each instance.

## Supporting information

BB-4015

## Testing instructions

I have yet to test this on stage.

The management script can be run manually and it will generate a CSV.

The Huey task can be tested by setting it to run more frequently (on stage?) and check whether the results are ok.

## Other information

It may make sense to refactor this, as the new periodic task is very similar of the existing one.
The crontab setting may need to be different so it does not reuse the exact same ones as the other report task.

### CODE REVIEW

File: reports/tasks.py
Comment (lgp171188) [inline reports/tasks.py:142]: @janimo, since this function is duplicating the task above, can you use something like the solution documented [here](https://stackoverflow.com/questions/51977211/how-to-create-a-task-function-with-an-argument-using-huey) to refactor the common code and just have a toggle that determines whether to restrict the queryset to the instances created in the last completed month? The code will become simpler with that change.

File: playbooks/collect_activity/stats.py
Comment (lgp171188) [inline playbooks/collect_activity/stats.py:83]: @janimo, this condition will never be true since `--start-date` and `--end-date` are required parameters? I'd recommend having the following logic:
* Both parameters should either be specified together or not at all.
* In the latter case, let the time range default to the last completed month.

Does this make sense?

File: instance/management/commands/instance_statistics_csv.py
Comment (lgp171188) [inline instance/management/commands/instance_statistics_csv.py:234]: @janimo, can you add/update the existing unit tests for this management command for your changes in this PR?

**Review by lgp171188 — CHANGES_REQUESTED** [review]
@janimo, I have reviewed this PR and left some review comments. Please check and address those. Also fix the failing unit tests before requesting another review.

File: reports/tasks.py
Comment (janimo) [reply reports/tasks.py:142]: @lgp171188 I agree that since much of the code is duplicated refactoring would make sense. Besides the queryset, the email subject and the error messages should be different.

File: playbooks/collect_activity/stats.py
Comment (janimo) [reply playbooks/collect_activity/stats.py:83]: @lgp171188 I overlooked the required=True attribute when copying the code, since there was a default=None setting. The logic you describe was my intention, and should work if the required=True is removed which I have done now. The args should not be required, in case the stats.py script is being called from elsewhere, but also for backwards compat while we develop.

**Comment by lgp171188** [comment]
@jani, [this failure](https://app.circleci.com/pipelines/github/open-craft/opencraft/3629/workflows/be6eedf6-1d04-453d-b5ab-11258bde10aa/jobs/32607) is due to the changes made for this task not having the corresponding updates to the integration test case in question. Can you check and fix it?

File: playbooks/collect_activity/stats.py
Comment (lgp171188) [inline playbooks/collect_activity/stats.py:93]: @janimo, this is a native datetime though the time zone support is active. You should use the UTC timezone for these datetime objects.

Comment (janimo) [reply playbooks/collect_activity/stats.py:93]: @lgp171188 you mean in the active_users filter? That is handled differently from the other date queries (like set of instances created in last month) because they run on the instances database and not on OCIM's own?

Is using utcnow() above not enough? Should it be datetime.now(tz=utc)? From what I see the last_login is stored in UTC, and so is I assume the time argument passed to the script.

File: reports/tasks.py
Comment (lgp171188) [inline reports/tasks.py:123]: @janimo, can you add/update the tests in `reports/tests.py` (`ReportTaskTestCase` class) for the changes made in this PR?

File: playbooks/collect_activity/stats.py
Comment (lgp171188) [reply playbooks/collect_activity/stats.py:93]: > The return value will give us the current date in UTC, right?

@janimo, a time zone has a meaning only for the `datetime` objects and not `date` objects.

**Comment by lgp171188** [comment]
@janimo, can you check and update the PR to address my open review comments? I will do another review pass and if everything is okay at that point, we can merge this PR.

File: reports/tasks.py
Comment (janimo) [reply reports/tasks.py:123]: I have now added a few tests in that file.

File: playbooks/collect_activity/stats.py
Comment (lgp171188) [reply playbooks/collect_activity/stats.py:93]: @janimo, this is still not addressed.

Comment (janimo) [reply playbooks/collect_activity/stats.py:94]: @lgp171188 I have fixed it to use tz aware datetimes instead.

**Review by lgp171188 — APPROVED** [review]
@janimo, :+1:. You can merge this PR **after** all the CI checks pass. If the integration tests fail due to some issue **not related** to the changes in this PR, please block this PR merge and the task progression on that fix. Otherwise, please fix the issues, if any, before merging.

- [x] I tested this on stage and verified that it works as expected.
- [x] I read through the code
- [x] ~~I checked for accessibility issues~~ NA
- [x] ~~Includes documentation~~

**Pull request merged**
